packages/api/endpoints/distribution.js

```diff
@@ -135,6 +137,33 @@ async function handleFileRequest(req, res) {
     return redirectToGetAuthorizationCode.send('Redirecting');
   }
 
+  req.authorizedMetadata = { userName: accessTokenRecord.username };
+  return next();
+}
+
+/**
+ * Responds to a request for temporary s3 credentials.
+ *
+ * @param {any} req - express request object
+ * @param {any} res - express response object
+ * @returns {Promise<Object>} the promise of express response object containing
+ * temporary credentials
+ */
+async function handleCredentialRequest(req, res) {
+  res.status(200);
```

Comment:
Seems like we shouldn't set `res.status` here but let `s3credentials` handle it since there is a try/catch in there where we could return 200 or a 500 error? Or am I missing something and this happens automatically?

Reply:
This might be bad refactoring.  Originally res was having it status set to 307 
https://github.com/nasa/cumulus/blob/master/packages/api/endpoints/distribution.js#L108-L137

And I used/copypastaed that originally to handle the redirect when I was figuring this out.  So You're probably right.  Let me see if I can get rid of that. I might have to modify the middleware to not set the status (which it probably shouldnt)

Reply:
Good catch, that was all kinda bad.  What about this? (I considered a`const temporaryRedirect = 307` but it didn't seem to make it clearer, fight me)

Reply:
13a9eae47